src/sql/workbench/browser/editor/resourceViewer/resourceViewerInput.ts

```diff
@@ -0,0 +1,97 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { TableDataView } from 'sql/base/browser/ui/table/tableDataView';
+
+import * as nls from 'vs/nls';
+
+import { EditorInput } from 'vs/workbench/common/editor';
+import { Event, Emitter } from 'vs/base/common/event';
+import { URI } from 'vs/base/common/uri';
+import { ResourceViewerState } from 'sql/workbench/common/editor/resourceViewer/resourceViewerState';
+
+export interface ColumnDefinition extends Slick.Column<Slick.SlickData> {
+	name: string;
+}
+
+export class ResourceViewerInput extends EditorInput {
+
+	public static ID: string = 'workbench.editorinputs.resourceviewerinputs';
+	public static SCHEMA: string = 'resource-viewer';
+	private _data: TableDataView<Slick.SlickData>;
+	private _columns: string[] = [];
+	private _state: ResourceViewerState;
+
+	private _onColumnsChanged = new Emitter<Slick.Column<Slick.SlickData>[]>();
+	public onColumnsChanged: Event<Slick.Column<Slick.SlickData>[]> = this._onColumnsChanged.event;
+
+	constructor() {
+		super();
+		this._state = new ResourceViewerState();
+		let searchFn = (val: { [x: string]: string }, exp: string): Array<number> => {
```

Comment:
It might be worth making an exported type in tableDataView for this type

Reply:
I'm going to revisit this once I get more implementation - I might actually make this more generic so that profiler/this/others can re-use the same functionality too.